app/src/components/adminConsole/manageTeam/SnappableManager.tsx

```diff
@@ -1,16 +1,23 @@
 import React, { useState, useRef, useCallback } from "react";
-import getSnappables from "../../firebase/users/GetSnappables";
-import { snappablesToCsvDownload, readFileAndUpload } from "./csvManager";
+import getSnappables from "../../../firebase/users/GetSnappables";
+import {
+    FileUploadWrapper,
+    SectionHeader,
+    SectionHeaderUnderline,
+} from "../AdminConsoleStyles";
+import { snappablesToCsvDownload, readFileAndUpload } from "../csvManager";
+import DownloadButton from "./DownloadButton";
 
 const LOADING = "loading";
 const IDLE = "idle";
 const ERROR = "error";
 
 const SnappableManager = () => {
     const [status, setStatus] = useState({ status: IDLE });
+    const [filename, setFilename] = useState<String | null>(null);
     const fileRef = useRef(null);
 
-    const downloadSnappables = useCallback(() => {
```

Comment:
For me, it is much clearer to put the useCallback in this function, rather than using it the component below (DownloadButton), any reason why you did this?
In addition, putting it in this function reduces the number of unnecessary rerenders.

Reply:
Or like, if you're going to take out the useCallback, take it out in DownloadButton as well.

Reply:
good point - I was just thinking about this. will change this 👍